sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -0,0 +1,242 @@
+      `The service returned ${ordinal} responses than inputs. Some errors may be treated as fatal.`
+    );
+  }
+
+  const result: U[] = [];
+  for (const sortedItem of sortedArray) {
+    const item = unsortedMap.get(sortedItem.id);
+    if (item) {
+      result.push(item);
+    }
+  }
+  return result;
+}
+
+/**
+ * @internal
+ */
+export interface AssessmentIndex {
+  document: number;
+  sentence: number;
+  assessment: number;
+}
+
+/**
+ * @internal
+ */
+export function parseAssessmentIndex(pointer: string): AssessmentIndex {
+  const regex = new RegExp(/#\/documents\/(\d+)\/sentences\/(\d+)\/assessments\/(\d+)/);
+  const res = regex.exec(pointer);
+  if (res !== null) {
+    const assessmentIndex: AssessmentIndex = {
+      document: parseInt(res[1]),
+      sentence: parseInt(res[2]),
+      assessment: parseInt(res[3]),
+    };
+    return assessmentIndex;
+  } else {
+    throw new Error(`Pointer "${pointer}" is not a valid Assessment pointer`);
+  }
+}
+
+/**
+ * Parses the index of the healthcare entity from a JSON pointer.
+ * @param pointer - a JSON pointer representing an entity
+ * @internal
+ */
+export function parseHealthcareEntityIndex(pointer: string): number {
+  const regex = new RegExp(/#\/results\/documents\/(\d+)\/entities\/(\d+)/);
+  const res = regex.exec(pointer);
+  if (res !== null) {
+    return parseInt(res[2]);
+  } else {
+    throw new Error(`Pointer "${pointer}" is not a valid healthcare entity pointer`);
+  }
+}
+
+/**
+ * Set the opinion mining property
+ * @internal
+ */
+export function setOpinionMining<X extends { includeOpinionMining?: boolean }>(
+  x: X
+): X & { opinionMining?: boolean } {
+  return { ...x, opinionMining: x.includeOpinionMining };
+}
+
+/**
+ * Set the pii categories property
+ * @internal
+ */
+export function setCategoriesFilter<X extends { categoriesFilter?: string[] }>(
+  x: X
+): X & { piiCategories?: string[] } {
+  return { ...x, piiCategories: x.categoriesFilter };
+}
+
+export function setSentenceCount<X extends { maxSentenceCount?: number }>(
+  x: X
+): X & { sentenceCount?: number } {
+  return { ...x, sentenceCount: x.maxSentenceCount };
+}
+
+export function setOrderBy<X extends { orderBy?: string }>(x: X): X & { sortBy?: string } {
+  return { ...x, sortBy: x.orderBy };
+}
+
+/**
+ * @internal
+ */
+export function addParamsToTask<X extends Record<string, any>>(
+  action: X
+): { parameters?: Omit<X, "actionName">; taskName?: string } {
+  const { actionName, ...params } = action;
+  return { parameters: params, taskName: actionName };
+}
+
+/**
+ * @internal
+ */
+export function compose<T1, T2, T3>(fn1: (x: T1) => T2, fn2: (y: T2) => T3): (x: T1) => T3 {
+  return (value: T1) => fn2(fn1(value));
+}
+
+/**
+ * @internal
+ */
+export async function getRawResponse<TOptions extends OperationOptions, TResult>(
+  f: (options: TOptions) => Promise<TResult>,
+  options: TOptions
+): Promise<LroResponse<TResult>> {
+  const { onResponse } = options || {};
```

Comment:
Could you just add a default param value `= {}` instead?

Reply:
FWIW, `TOptions` might have non-optional stuff.

Reply:
Makes sense.